### PR TITLE
Example scripts overhaul 

### PR DESCRIPTION
**Pull request by HYPExMon5ter**
Target Minecraft versions: Any
Requirements: None
Related issues: #1201 

Description:
I recoded most of the example scripts that get generated upon generation of the Skript folder.

### CODE REVIEW

File: src/main/resources/scripts/drop fixes.sk [hunk 1]
```diff
@@ -2,13 +2,12 @@
...
-	drop glass pane
-
-on break of bookshelf:
-	cancel event
-	set block to air
-	drop bookshelf
+on break:
+	if event-block is glass:
+		drop glass
+	else if event-block is glass pane:
```
Comment (Snow-Pyon) [inline src/main/resources/scripts/drop fixes.sk]: I wouldn't say this change is actually good. Keeping the triggers modularized isn't a bad practice.

File: src/main/resources/scripts/equip anything.sk [hunk 2]
```diff
@@ -5,18 +5,9 @@
 aliases:
 	custom helmets = iron block, gold block, diamond block # Change this to anything you'd like
 
-on inventory click:
```
Comment (Snow-Pyon) [inline src/main/resources/scripts/equip anything.sk]: The whole point of the example script was this part, why did you remove it?

File: src/main/resources/scripts/furnace automatisation.sk [hunk 3]
```diff
@@ -12,32 +12,29 @@ options:
...
-	loop blocks in radius {@radius}:
-		loop-block is chest
-		loop items of type fuel:
-			loop-block contains loop-item
-			remove loop-item from loop-block
-			set fuel of event-block to loop-item
-			stop trigger
+	if fuel slot of block will be empty:
+		loop blocks in radius {@radius}:
+			if loop-block is a chest:
```
Comment (Snow-Pyon) [inline src/main/resources/scripts/furnace automatisation.sk]: the old format was better in this specific case, using in-line in this kind of case is important.

File: src/main/resources/scripts/homes.sk [hunk 4]
```diff
@@ -3,20 +3,38 @@
...
+			send "Successfully deleted your home '%arg-1%'."
+		else:
+			send "You don't have a home with that name."
+
+command /home <text>:
+	description: Teleport yourself to your home.
+	permission: skript.home
+	executable by: players
+	trigger:
+		if {homes.%player%::%arg-1%} is set:
```
Comment (Snow-Pyon) [inline src/main/resources/scripts/homes.sk]: `{homes::%player%::%arg-1%}` would be preferred even if it the same thing.

File: src/main/resources/scripts/plant with hoe.sk [hunk 5]
```diff
@@ -9,24 +9,15 @@ aliases:
...
-# right click sand to plant a cactus
-on rightclick on sand with any hoe:
-	player has a cactus
-	block above is air
-	blocks north above, east above, south above and west above are allowed next to cactus
-	
-	damage tool by 1
-	set block above to cactus
-	remove a cactus from player
+on right click:
```
Comment (Snow-Pyon) [inline src/main/resources/scripts/plant with hoe.sk]: Same thing as above, keeping triggers modularized is good.

File: src/main/resources/scripts/simple motd.sk [hunk 6]
```diff
@@ -20,4 +20,5 @@ command /showmotd:
 on join:
 	# uncomment the following line to make the MOTD appear after other messages
 	# wait a tick
-	message {motd}
+	if {motd} is set:
```
Comment (Snow-Pyon) [inline src/main/resources/scripts/simple motd.sk]: useless check, the message won't be sent if the variable isn't set.

File: src/main/resources/scripts/equip anything.sk (hunk 2 above)
Comment (HYPExMon5ter) [reply src/main/resources/scripts/equip anything.sk]: it didn't work so I removed it as I couldn't figure out how to fix it.

File: src/main/resources/scripts/furnace automatisation.sk (hunk 3 above)
Comment (HYPExMon5ter) [reply src/main/resources/scripts/furnace automatisation.sk]: This script doesn't work properly anyway, I didn't really attempt fixing it either, I'll look into it and revert to the old format.

File: src/main/resources/scripts/homes.sk (hunk 4 above)
Comment (HYPExMon5ter) [reply src/main/resources/scripts/homes.sk]: Will convert to uuid and new variable.

File: src/main/resources/scripts/plant with hoe.sk (hunk 5 above)
Comment (HYPExMon5ter) [reply src/main/resources/scripts/plant with hoe.sk]: what do you mean? like `on right click of grass, dirt or soil:`?

File: src/main/resources/scripts/drop fixes.sk (hunk 1 above)
Comment (HYPExMon5ter) [reply src/main/resources/scripts/drop fixes.sk]: will revert

File: src/main/resources/scripts/furnace automatisation.sk (hunk 3 above)
Comment (HYPExMon5ter) [reply src/main/resources/scripts/furnace automatisation.sk]: I've tried fixing this with @Pikachu920 and we can't get it to set the fuel in the furnace, so idk what to do.

File: src/main/resources/scripts/equip anything.sk (hunk 2 above)
Comment (bensku) [reply src/main/resources/scripts/equip anything.sk]: How does it not work... ?

File: src/main/resources/scripts/plant with hoe.sk (hunk 5 above)
Comment (bensku) [reply src/main/resources/scripts/plant with hoe.sk]: Yes. Latest release has them bugged. That is not a reason to change example scripts.

File: src/main/resources/scripts/furnace automatisation.sk [hunk 7]
```diff
@@ -15,14 +15,15 @@ options:
 	radius: 1
 
 on fuel burn:
-	fuel slot of block will be empty
+	fuel slot of event-block will be empty
+	wait 1 second
 	loop blocks in radius {@radius}:
-		loop-block is chest
-		loop items of type fuel:
```
Comment (bensku) [inline src/main/resources/scripts/furnace automatisation.sk]: How it this an improvement? Inventories are most commonly larger than amount of fuels that Minecraft has.

File: src/main/resources/scripts/furnace automatisation.sk [hunk 8]
```diff
@@ -15,14 +15,15 @@ options:
 	radius: 1
 
 on fuel burn:
-	fuel slot of block will be empty
+	fuel slot of event-block will be empty
+	wait 1 second
```
Comment (bensku) [inline src/main/resources/scripts/furnace automatisation.sk]: If this doesn't work without a delay, you should just open an issue... It is not ok to circumvent bugs in ugly ways in example scripts.

File: src/main/resources/scripts/furnace automatisation.sk [hunk 9]
```diff
@@ -32,12 +33,12 @@ on smelt:
 			loop-block contains loop-item
 			remove loop-item from loop-block
 			set ore of event-block to loop-item
-			stop trigger
+			stop
```
Comment (bensku) [inline src/main/resources/scripts/furnace automatisation.sk]: Stop trigger is more clear.

File: src/main/resources/scripts/kill counter.sk [hunk 10]
```diff
@@ -2,25 +2,24 @@
 # Counts kills per life, in total, and the highest kill streak.
 #
 
-# Defaults are useful here, as a player's kills would be '<none>' if he hasn't killed anything yet,
-# but with defaults defined those will be used in that case.
-variables:
-	{kill counter.%player%.kills_session} = 0
```
Comment (bensku) [inline src/main/resources/scripts/kill counter.sk]: This introduces a potentially useful feature. The fact that docs do not explicitly mention it... Well, not an issue in example scripts. Docs should be fixed, not this.

File: src/main/resources/scripts/teleport with compass.sk [hunk 11]
```diff
@@ -8,9 +8,8 @@ aliases:
 	nonsolid = 0, 6, 27, 28, 30, 31, 32, 37, 38, 39, 40, 50, 55, 59, 63, 64, 65, 66, 68, 69, 70, 71, 72, 75, 76, 77, 78:0, 83, 85:4-7, 104, 105, 106, 111, 115, 131, 132, 141, 142, 143, 147, 148, 157
 
 on rightclick with compass:
-	player has permission "skript.teleport"
-	loop blocks above targeted block:
-		loop-block is nonsolid
-		block above loop-block is nonsolid
-		teleport player to loop-block
-		stop trigger
+	if player has permission "skript.teleport":
```
Comment (bensku) [inline src/main/resources/scripts/teleport with compass.sk]: It is allowed and in this case even recommended to not indent and not to have keyword if.

File: src/main/resources/scripts/simple god mode.sk [hunk 12]
```diff
@@ -3,6 +3,6 @@
 #
 
 on damage:
-	victim is a player
-	victim has permission "skript.god"
-	cancel event
```
Comment (bensku) [inline src/main/resources/scripts/simple god mode.sk]: Old code was perfectly valid. Actually I consider it better. Nesting if sections looks ugly and increases line width of script. In this case, it is also useless.

File: src/main/resources/scripts/kill counter.sk [hunk 13]
```diff
@@ -2,25 +2,24 @@
...
+		set {killcounter::%attacker%::kill-streak} to 0
+		set {killcounter::%attacker%::kills} to 0
+	if attacker is a player:
+		add 1 to {killcounter::%attacker%::kills}
+		add 1 to {killcounter::%attacker%::kills-this-life}
+		if {killcounter::%attacker%::kills-this-life} is greater than {killcounter::%attacker%::kill-streak}:
+			set {killcounter::%attacker%::kill-streak} to {killcounter::%attacker%::kills-this-life}
 
 on death of player:
-	set {kill counter.%player%.kills_session} to 0
```
Comment (bensku) [inline src/main/resources/scripts/kill counter.sk]: Using spaces in variable names is encouraged if it looks cleaner that way and/or when it is grammatically correct.

File: src/main/resources/scripts/equip anything.sk [hunk 14]
```diff
@@ -5,18 +5,9 @@
...
-	inventory action is place all or nothing
-	cursor slot of player is custom helmets # Check if item is allowed as helmet
-	event-slot is helmet slot of event-player # Check that player clicked the helmet
-	cancel the event
-	set {_old helmet} to helmet of player # Store old helmet
-	set helmet of player to cursor slot of player # Set new helmet to item in player's cursor
-	set cursor slot of player to {_old helmet} # Set player's cursor to old helmet, possibly air
-
 on right click:
-	tool of player is custom helmets
```
Comment (bensku) [inline src/main/resources/scripts/equip anything.sk]: Nesting ifs is a bad practise if it is not needed.

File: src/main/resources/scripts/equip anything.sk (hunk 2 above)
Comment (HYPExMon5ter) [reply src/main/resources/scripts/equip anything.sk]: idk, it just didn't, atleast on the latest release, I didn't wanna test on a older release (dev33) incase of fixed bugs or something.

File: src/main/resources/scripts/plant with hoe.sk (hunk 5 above)
Comment (HYPExMon5ter) [reply src/main/resources/scripts/plant with hoe.sk]: alright, i'll change it back, I just didn't want it erroring on the latest release.

File: src/main/resources/scripts/furnace automatisation.sk (hunk 8 above)
Comment (HYPExMon5ter) [reply src/main/resources/scripts/furnace automatisation.sk]: I forgot to remove the one second wait, it doesn't work with or without it.

File: src/main/resources/scripts/furnace automatisation.sk (hunk 9 above)
Comment (HYPExMon5ter) [reply src/main/resources/scripts/furnace automatisation.sk]: will revert

File: src/main/resources/scripts/kill counter.sk (hunk 10 above)
Comment (HYPExMon5ter) [reply src/main/resources/scripts/kill counter.sk]: so you want me to add the variables section back?

File: src/main/resources/scripts/simple god mode.sk (hunk 12 above)
Comment (HYPExMon5ter) [reply src/main/resources/scripts/simple god mode.sk]: Same as above, will revert

File: src/main/resources/scripts/kill counter.sk (hunk 13 above)
Comment (HYPExMon5ter) [reply src/main/resources/scripts/kill counter.sk]: Okay, I'll fix it.

File: src/main/resources/scripts/equip anything.sk (hunk 14 above)
Comment (HYPExMon5ter) [reply src/main/resources/scripts/equip anything.sk]: Will revert

File: src/main/resources/scripts/furnace automatisation.sk (hunk 7 above)
Comment (HYPExMon5ter) [reply src/main/resources/scripts/furnace automatisation.sk]: It didn't work before, so I recoded it this way, and it still doesn't work, but it works more than before, it atleast removes the fuel from the chest now.

File: src/main/resources/scripts/equip anything.sk (hunk 2 above)
Comment (bensku) [reply src/main/resources/scripts/equip anything.sk]: That would be a rather high priority bug, then. Ouch.

Comment (HYPExMon5ter) [reply src/main/resources/scripts/equip anything.sk]: I tried it on dev33 as well and it didn't work, so what do you want me to do? Open an issue or what?

Comment (HYPExMon5ter) [reply src/main/resources/scripts/equip anything.sk]: Will do.

Comment (HYPExMon5ter) [reply src/main/resources/scripts/equip anything.sk]: Created #1210 .

**Comment by HYPExMon5ter** [comment]
@bensku what about the furnace automisation script? Should I open an issue for that too or what?

**Comment by bensku** [comment]
Sorry for slow response. Yes: open issues for everything in example scripts that should work, but doesn't.

**Comment by HYPExMon5ter** [comment]
alright, I created #1242 and #1243

**Comment by Nicofisi** [comment]
What's the status of this PR?

**Comment by HYPExMon5ter** [comment]
Well, need to figure out #1210 and #1243 first

**Comment by Snow-Pyon** [comment]
I am so sorry but the changes you made are not exactly what we're looking for, I decided to close the PR since it's been a while and there are things that could be handled differently.

Again, I apologize for denying it but I hope you'll understand it.
